### ISSUE TITLE
Title for data frame and Instat Object

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#3741]
This issue concerns the metadata for the data frame and also for the Instat object.
My background is to include information on case studies, from the ILRI Biometrics set.
1. They currently have a little Word file corresponding to each Excel file - with the data.
2. This has descriptive information about the file (often 3 or 4 lines.  That's the meatdata for the data frame.
3.  There is then information on each column.  That's the metadata at the column level, i.e. the Variable description.  That's OK with what we have.
4. There are usually between 1 and 4 Excel files (data frames) for each study.  I propose to put them together into an Instat Object.  

I currently have two problems.
1. I can add the data frame description.  There is a field for that already.  But it isn't easily visible if I have long texts there.  Genstat has a special menu item for the "Sheet title" - shown below.  Should we do anything more in R-Instat?
2. There isn't anywhere yet for a "file title/description".

![image](https://user-images.githubusercontent.com/11226469/28772236-789cc246-75dd-11e7-8a15-c8473b8777c1.png)

In general these descriptions could be quite long.  They might include a field plan, or even link to a video.  What should we include in R-Instat and how should we add links - if needed, to (say) pdf, or pictures or videos, or ...

**Comment by dannyparsons** [comment]
How about a dialog for Description? 
This could be similar to above but with a selector for choosing a column or just a data frame. There is then a large multi line text box for displaying and editing the current 'label' property.
This could be on the menu or accessible as a right click somewhere in the grids.

**Comment by rdstern** [comment]
That would be good.
But there are 3 "levels" - so these nice buttons?
a) **Column** - where there currently is the name and the label in the metadata - so this would be an alternative.  This is also available with the rename dialogue.
b) **Dataframe** - where this can currently be edited in the sheet metadata.  But, unlike rename there is no dialogue.  I like the idea of the multi-line field here.
c) **File/Instat Object** where we don't currently have a way of entering (or viewing) any metadata.

**Comment by dannyparsons** [comment]
Yes that's right we could have all 3, at the moment we only store the first 2.
So I will think out a dialog for this for 0.4.2.

**Comment by rdstern** [comment]
We already have **Prepare > Data Frame > Rename**, which is to rename a column and also the label.  This could change simply to emphasise that the label is usually longer than the name.
We also already have **Prepare > Data Object > Rename Data Frame**.  This permits the data frame to be renamed and also a label to be added.  This label field could be made bigger and also multi-line.
Then either we have a third simple dialogue, namely **Prepare > Data Object > Rename Instat Object**.  This is the same, but at the file level.  It may just have the description, so not the name of the obhject itself?
Or we have a single dialogue, namely **Prepare > Data Object > Rename** and have our famous buttons at the top, with **Data Frame** or **Instat Object** as the 2 options.

There is a related issue - so I put it here, though I am sure David would want it to be separate.  This is the display of the labels.  For example SPSS permits our data selector to show either the column names or the labels.  Some of our analyses do use the column labels when they exist.
But nothing currently uses the data frame or instat object labels.  So the dialogue should (at least) have an option  to display them in the output window.
